### DISCUSSION TITLE
버튼을 누르면 텔레포트 하며 방향 Rotation 전환법

### DISCUSSION THREAD

**Discussion by 3ANAKO** [naverz/zepeto-studio-kor#995] [5. Q&A]
안녕하세요 
유니티를 처음하는데 문제가 생겨 질문합니다

가까이 가서 버튼을 누르면 의자에 앉고 이동하면 풀리는 것을 만들고 있는데 

버튼 누르고 텔레포트후 애니메이션 출력까진 되는데 동시에 방향 전환법을 모르겠네요.

유니티가 처음이라 어찌어찌 짜집기 하면서 여기까지 만들었는데 
이건 잘 모르겠네요. 스크립트 올릴게요 답좀 알려주세요!

![11](https://user-images.githubusercontent.com/109119009/178615277-ae7b0cb5-2749-42c1-8454-227ad0e2bc7b.png)

**Comment by YSHFM** [comment]
qaternion.identity는 0,0,0,1인 쿼터니언 값을 반환합니다.
30행에서 qaternion.identity값을 다른 쿼터니언 값을 줘도 회전이 안되신다면 
gameobject.transform.rotation에 쿼터니언 값을 주시거나,
gameobject.transform.localrotation에 쿼터니언 값을 주시면 됩니다.
localrotation과 rotation차이는 로컬 좌표계 회전값과 월드 좌표계 회전값으로 이해해주시면 되겠습니다.

**Reply by 3ANAKO** [reply]
this.zepetoCharacter.Teleport(new Vector3(13.8, -0.61, 8.6), Quaternion.identity(0, 0, 90, 1));
30행 이런식으로 넣으면 되나요? 
이렇게 했는데 "This expression is not callable. Type 'Quaternion' has no call signatures' 이라 뜨네요.. 잘모르겠어요...

**Reply by YSHFM** [reply]
https://showmiso.tistory.com/57

쿼터니언은 오일러각과는 다른 좌표계입니다. inspector에 보이는 각도는 오일러각이며 쿼터니언과는 다릅니다.
오일러각을 쿼터니언으로 변환하려면  UnityEngine.Quaternion.Euler(x,y,z)를 사용하면 쿼터니언 자료형으로 반환됩니다.

**Reply by YSHFM** [reply]
https://docs.unity3d.com/kr/2020.3/ScriptReference/Quaternion-identity.html
 Quaternion.identity에 대해 참고할 만한 공식 레퍼런스입니다.

**Reply by 3ANAKO** [reply]
감사합니다 쿼터니언 이라는 개념을 처음 알았네요 
많은 도움 되었어요
